src/python/alias.cpp

```diff
@@ -112,23 +116,56 @@ static void set_variant(nb::args args) {
     }
 
     if (!curr_variant.equal(new_variant)) {
-        curr_variant = new_variant;
-        nb::object curr_variant_module = variant_module(curr_variant);
+        nb::object new_variant_module = variant_module(new_variant);
 
-        nb::dict variant_dict = curr_variant_module.attr("__dict__");
+        nb::dict variant_dict = new_variant_module.attr("__dict__");
         for (const auto &k : variant_dict.keys())
             if (!nb::bool_(k.attr("startswith")("__")) &&
                 !nb::bool_(k.attr("endswith")("__")))
                 Safe_PyDict_SetItem(mi_dict, k.ptr(),
                     PyDict_GetItem(variant_dict.ptr(), k.ptr()));
 
+        const auto &callbacks = nb::borrow<nb::set>(variant_change_callbacks);
+        for (const auto &cb : callbacks)
+            cb(curr_variant, new_variant);
+
+        // TODO: replace this with a callback?
```

Comment:
We could use the opportunity to remove this special case and add a callback in `mitsuba.python.ad.integrators.__init__`. But I wasn't sure how to test the new behavior is correct.

Does the test suite check / relies on this reload to correctly happen?